### ISSUE TITLE
CancelProcessInstanceBanTest#shouldBanInstanceWhenTerminatingInstanceWithALotOfNestedChildInstances

### ISSUE THREAD

**Issue by npepinpe** [camunda/zeebe#13297]
**Summary**

> Try to answer the following as best as possible

- How often does the test fail? 5 times today :upside_down_face: 
- Does it block your work? No, but it's the flakiest thing around I think
- Do we suspect that it is a real failure? Can't say

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
CancelProcessInstanceBanTest.shouldBanInstanceWhenTerminatingInstanceWithALotOfNestedChildInstances:61 ? StreamWrapper No event found matching the criteria
</pre>
</details>

**Logs**

https://gist.github.com/npepinpe/20d2da78b2af4174767fe5cd407b08f2

**Comment by korthout** [comment]
@remcowesterhoud I've encountered this one a few times today as well.

I've played around with different ideas:
- increasing the nesting depth to ridiculous numbers (10000+) -> still flaky
- moving the test case to an IT using the EmbeddedBrokerRule -> still flaky

In addition to its instability, the test takes exceptionally long to run: ~5s.

Is this test really necessary? It might be better to remove it in its current form.

Instead, I see options:
- ask ZDP to add a test case that makes sure the StreamPlatform can handle the Engine throwing `java.lang.VirtualMachineError` like the `java.lang.StackoverflowException` (and perhaps even other `java.lang.Error`s).
- we can catch `java.lang.Error` at the highest level in the Engine, and transform it to a recoverable exception so the StreamPlatform provides it to us via the normal error handling.

@remcowesterhoud WDYT?
